### ISSUE TITLE
Regression: MimeTypeDetector no longer identifies content if there is no file extension

### ISSUE THREAD

**Issue by alandavis** [reportportal/client-java#234]
**Describe the bug**
Having upgraded to 5.2.0 (also happens in 5.2.1.and 5.2.2) from 5.1.27 the `MimeTypeDetector.detect(file)` method no longer identifies the mimetype by looking at the content of the file.

**Steps to Reproduce**
1. Start with a JPEG file and rename it to have no extension. For example `/tmp/file3146527651826023179tmp`.
2. Call the method.

**Expected behavior**
The method should return `image/jpeg` as it did previously in 5.1.27.

**Actual behavior**
The method returns `application/octet-stream` as it has failed to identify the MIME type.

**Dependency versions**
5.2.0 to 5.2.2 have regressed since 5.1.27. 

**Additional context**

**Comment by HardNorth** [comment]
@alandavis There were no changes in this method since 2021. And recent changes (5.2.3) are related to resource location, not to detect method: https://github.com/reportportal/client-java/blame/develop/src/main/java/com/epam/reportportal/utils/MimeTypeDetector.java#L57-L86

Actually this type of detection relies on `URLConnection.guessContentTypeFromStream` method, which is basic JRE method. I guess with the migration to 5.2.x client you also changed JDK/JRE vendor and this version of API detects MIME types worse than previous.

**Comment by alandavis** [comment]
Thanks for coming back so quickly.

The method did change slightly between 5.1.27 and 5.2.0. Line 58 previously was:
> 		String type = URLConnection.guessContentTypeFromStream(new ByteArrayInputStream(Utils.readInputStreamToBytes(new FileInputStream(
> 				file))));

Initially I thought this was not significant but I can confirm the bytes at the start of the file in question are `ff d8 ff e0` so should have been identified by `guessContentTypeFromStream` as `"image/jpeg"`. Could this be a little/big endian issue https://en.wikipedia.org/wiki/Endianness ? Our code is being run in a GitHub Action and the agent identifies itself as `Ubuntu 22.04.3 LTS Image: ubuntu-22.04 2024-02-06T18:32:24.4719451Z Version: 20240201.1.0`

Unfortunately the only change is the version of this jar. The JDK/JRE are the same. JDK version 17 from `corretto 17.0.10-7.1/x64`

I think we can discount the big change going from 5.1.27 to 5.20 which was the removal of `com.google.guava:guava:jar:30.1-android:compile` as the detection still works with 5.1.27 with that jar excluded. The reason I was looking in there was that it might have contained `FileTypeDetectors` which would be used if the code reached `probeContentType` (line 60).

**Comment by HardNorth** [comment]
@alandavis I tried reading with different methods, but it all the same, unfortunately. I can add additional internal byte detection method, hope it will help.

**Comment by HardNorth** [comment]
@alandavis That's all I can do: https://github.com/reportportal/client-java/releases/tag/5.2.4

Basically, I duplicated some logic for jpeg and png detection from JDK and extended it a bit.

BTW: While working on this release I found that not all methods of `IOUtils.copy` of `commons-io` library copy file "as is", some of them do that through char array, messing binary data on the way. I hit one of such methods while implementing the release and spend around hour figuring that out. So I implemented my own copy method, to not get confused anymore. Maybe this might help you also.
